### PR TITLE
add mapping for ssh property in build section

### PR DESCRIPTION
**Pull request by glours**
This implement https://github.com/compose-spec/compose-spec/pull/234

### CODE REVIEW

File: loader/loader_test.go [hunk 1]
```diff
@@ -1867,3 +1867,59 @@ func TestLoadServiceWithVolumes(t *testing.T) {
...
+	assert.ErrorContains(t, err, "services.test.build.ssh must be a mapping")
+}
+
+func TestLoadSSHWithDefaultValueInBuildConfig(t *testing.T) {
+	actual, err := loadYAML(`
+services:
+  test:
+    build:
+      context: .
+      ssh: ssh-agent
```
Comment (ndeloof) [inline loader/loader_test.go:1901]: according to the spec proposal, should be `default`

File: types/types.go [hunk 2]
```diff
@@ -426,6 +427,9 @@ func (l Labels) Add(key, value string) Labels {
 	return l
 }
 
+// SSHConfig is a mapping type for SSH build config
+type SSHConfig map[string]string
```
Comment (ndeloof) [inline types/types.go:431]: I wonder `SSHConfig` should be a `[]SSHKey`, the latter declared as struct { ID, Path }` (to match the spec)

**Review by ndeloof — APPROVED** [review]

**Comment by ndeloof** [comment]
ready to merge, just waiting for https://github.com/compose-spec/compose-spec/pull/234 to be approved

**Comment by hmajid2301** [comment]
Is this all good to merge now ? The spec has been approved ?

**Pull request merged**